File: mobileBE/src/main/java/com/uth/mobileBE/dto/response/LoanResponse.java
package com.uth.mobileBE.dto.response;

import com.fasterxml.jackson.annotation.JsonFormat;
import com.uth.mobileBE.models.enums.StatusLoan;
import lombok.Builder;
import lombok.Data;
import java.time.LocalDateTime;
import java.util.List;

@Data
@Builder
public class LoanResponse {
    private Long loanId;
    private String libraryName;
    private String readerName;
    private String processorName;

    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd HH:mm:ss")
    private LocalDateTime borrowDate;

    private StatusLoan status;
    private List<String> bookTitles;

    // THÊM 2 TRƯỜNG NÀY VÀO:
    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd HH:mm:ss")
    private LocalDateTime createdAt;

    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd HH:mm:ss")
    private LocalDateTime updateAt;
}

File: app/src/main/java/com/example/quanlythuvien/data/model/response/LoanResponse.kt
package com.example.quanlythuvien.data.model.response

import com.google.gson.annotations.SerializedName

data class LoanResponse(
    @SerializedName("loanId") val loanId: Long,
    @SerializedName("libraryName") val libraryName: String,
    @SerializedName("readerName") val readerName: String,
    @SerializedName("processorName") val processorName: String?,
    @SerializedName("borrowDate") val borrowDate: String,
    @SerializedName("status") val status: String,

    // ĐỔI TÊN: Từ bookDetails thành loanDetails, đổi kiểu dữ liệu thành LoanDetailResponse
    @SerializedName("loanDetails") val loanDetails: List<LoanDetailResponse>?,

    @SerializedName("createdAt") val createdAt: String,
    @SerializedName("updateAt") val updateAt: String
)